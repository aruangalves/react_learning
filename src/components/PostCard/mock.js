export const postCardMock = {
    title: 'post title #1',
    body: 'some text for post #1',    
    cover: 'img/img.jpg'
};