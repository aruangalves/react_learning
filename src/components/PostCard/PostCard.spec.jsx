import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import { postCardMock } from "./mock";

const props = postCardMock;

describe('<PostCard />', ()=>{
    it('should render', ()=>{
        render(<PostCard {...props} />);

        expect(screen.getByRole('img', {name: props.title})).toBeInTheDocument();

        expect(screen.getByAltText(props.title)).toHaveAttribute('src', props.cover);

        expect(screen.getByRole('heading', {name: props.title})).toBeInTheDocument();

        expect(screen.getByText(props.body)).toBeInTheDocument();
    });

    it('should match snapshot', ()=>{
        const {container} = render(<PostCard {...props} />);

        const {firstChild} = container;
        
        expect(firstChild).toMatchSnapshot();

    });
});