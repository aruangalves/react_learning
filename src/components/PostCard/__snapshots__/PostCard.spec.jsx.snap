// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="post title #1"
    src="img/img.jpg"
  />
  <div
    class="post-content"
  >
    <h2>
      post title #1
    </h2>
    <p>
      some text for post #1
    </p>
  </div>
</div>
`;
